taf/repository_tool.py

```diff
@@ -0,0 +1,221 @@
+import getpass
+import json
+import os
+import shutil
+from contextlib import contextmanager
+from pathlib import Path
+import securesystemslib
+
+role_keys_cache = {}
+
+class Repository:
+
+  def __init__(self, repository, repository_path):
+    self._repository = repository
+    self.repository_path = repository_path
+
+  @property
+  def targets_path(self):
+    return os.path.join(self.repository_path, 'targets')
+
+  @property
+  def metadata_path(self):
+    return os.path.join(self.repository_path, 'metadata')
+
+  @property
+  def metadata_staged_path(self):
+    return os.path.join(self.repository_path, 'metadata.staged')
+
+  def add_existing_target(self, file_path, targets_role='targets'):
+    """
+    Registers new target files with TUF. The files are expected to be
+    inside the targets directory.
+    """
+    targets = self._role_obj(targets_role)
+    targets.add_target(os.path.join(self.targets_path, file_path))
+
+
+  def add_targets(self, data, targets_role='targets', files_to_keep=['repositories.json']):
```

Comment:
File `repositories.json` should always exist (used in _repositoriesdb.py_), so, in my opinion `files_to_keep` should be a constant or field in `Repository` class. E.G. Passing an empty array will break something probably, right?

Reply:
Yes, `repositories.json` should always exist. We need to consider two things though. There might be a need to update `repositories.json`, in which case we don't want to keep the old version. Secondly, someone using TAF might want to handle another file in a similar way. I guess that we could have a field which lists all files required by the framework, and still allow the users to specify additional files when calling `add_targets`, depending on their needs at a particular time. If `repositories.json` is also in `data`, we won't copy the previous version

Reply:
Maybe we need more methods to work with `repositories.json` file since it is a part of framework (in the future)? I can't think of example where user would want another "special" target file because we already have `additional_info` field to store custom data.

Reply:
I think that it's a good idea to keep this flexible. files_to_keep should contain a list of all files which should remain targets, but should not be modified. Meaning that there is no need to read their content and pass it as an item in the data dictionary. I thought that this option would be useful. The targets do not even have to have any particular importance.
I like the idea of adding methods for working with repositories.json. I'll create an issue